event-log/src/main/scala/io/renku/eventlog/subscriptions/SubscribersRegistry.scala

```diff
@@ -102,14 +102,12 @@ private class SubscribersRegistry(
 
   def subscriberCount(): Int = busyPool.size() + availablePool.size()
 
-  private def start(): IO[Fiber[IO, Unit]] = {
-    for {
-      _                        <- timer sleep checkupInterval
-      subscribersDueForCheckup <- findSubscribersDueForCheckup
-      _                        <- bringToAvailable(subscribersDueForCheckup)
-      _                        <- start()
-    } yield ()
-  }.start
+  private def busySubscriberCheckup(): IO[Unit] = for {
+    _                        <- timer sleep checkupInterval
+    subscribersDueForCheckup <- findSubscribersDueForCheckup
+    _                        <- bringToAvailable(subscribersDueForCheckup)
+    _                        <- busySubscriberCheckup()
```

Comment:
What happens if this method gets called from different threads? Also, could we start this method name with a verb? checkOnBusySubscribers or something.

Reply:
As it's a private method I think we've got control when it's called.